### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by zaccee** [hashicorp/terraform-provider-aws#35074]
### Terraform Core Version

1.6.6

### AWS Provider Version

5.31.0

### Affected Resource(s)

* aws_db_option_group

### Expected Behavior

Given that this resource exists on aws and I was able to generate a resource using the option group identifier, I should be able to import it into terraform.

### Actual Behavior

Cannot execute plan due to colon `:` in name.

### Relevant Error/Panic Output Snippet

```shell
│ Error: only lowercase alphanumeric characters and hyphens allowed in "name"
│ 
│   with aws_db_option_group.default_postgres_10,
│   on generated.tf line 3394, in resource "aws_db_option_group" "default_postgres_10":
│ 3394:   name                     = "default:postgres-10"
│ 
```
```


### Terraform Configuration Files

```
# __generated__ by Terraform
resource "aws_db_option_group" "default_postgres_10" {
  engine_name              = "postgres"
  major_engine_version     = "10"
  name                     = "default:postgres-10"
  name_prefix              = null
  option_group_description = "Default option group for postgres 10"
  tags                     = {}
  tags_all                 = {}
}
```

### Steps to Reproduce

1. Create postgres db in rds manually, and then use terraform import to import resources. Or just create a db option group with something like the hcl above.
```
import {
  id = "default:postgres-10"
  to = aws_db_option_group.default_postgres_10
}
```
2. terraform plan

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

The DB was pre-existing, trying to import everything into terraform that has been created by point-and-click.



### References

Related to this issue which only resolved for aws db parameter groups: https://github.com/hashicorp/terraform-provider-aws/issues/12144

### Would you like to implement a fix?

None

**Comment by acwwat** [comment]
I believe that the `default` namespace is reserved by AWS for option groups that are automatically created when you deploy an RDS instance of a particular engine. The validation contract is correct and it is enforced in both the GUI and the [API](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateOptionGroup.html).

Regardless, if you have a reason to customize an option group, you should create your own and associate it with your RDS DB instances. Otherwise just let AWS create it for you as you create your RDS DB instances (via GUI, Terraform, etc.)

**Comment by justinretzolk** [comment]
Given the information above, I suspect that this is expected behavior, however, I'd like to give the team/community a chance to investigate whether this is possible and something we should support.

**Comment by zaccee** [comment]
@acwwat Thanks for the information. So just to clarify, there is no need to import existing `aws_db_option_group` resources if your configuration is only using those from the `default` namespace? In such cases, these option groups will automatically be created by AWS when you create a new RDS DB instance?

**Comment by acwwat** [comment]
@zaccee Correct. Consider the following sample TF config which I just ran to be 100% certain:

```
resource "aws_db_instance" "example" {
  allocated_storage    = 20
  engine               = "postgres"
  engine_version       = "16"
  instance_class       = "db.t3.micro"
  db_name              = "mydb"
  username             = "foo"
  password             = "foobarbaz"
  skip_final_snapshot  = true
  db_subnet_group_name = "test"
}
```

When deploying an RDS for PostgreSQL instance at v16 for the first time in my AWS account using Terraform, AWS will automatically created the default option group `default:postgres-16` and the default parameter group `default.postgres16`. Just note that you cannot change any setting in these default option/parameter groups. You must create your own if you need to customize even one setting in either of them.
